cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -2860,6 +2860,38 @@
 (rule 3 (fnmadd ty (fneg x) y z) (fmadd ty x y z))
 (rule 4 (fnmadd ty x (fneg y) z) (fmadd ty x y z))
 
+
+(rule 2 (lower (has_type ty (fma x y (fneg z))))
+      (if-let $true (use_fma))
+      (fmsub ty x y z))
+
+;; fmsub and fnmsub
+(decl fmsub (Type Value Value Value) Xmm)
+(decl fnmsub (Type Value Value Value) Xmm)
+
+;; Base case, will sink a load of `z` automatically.
+(rule (fmsub ty x y z) (x64_vfmsub213 ty x y z))
```

Comment:
I forget what the details of our lowering and ISLE conversions are these days, but does this need to guard on sinkable loads? Or will it force things to registers as written? (In which case the comment should be updated)

cc @jameysharp, who I think would know the answer to this question.

Reply:
I'm not actually sure. But I see that this is identical to the existing lowering rules above aside from replacing `add` with `sub` and so on. So it's probably fine?

Reply:
I did just work off of the rules for the `add` instructions. However, its probably related to the semantics of `z` matching into `XmmMem` in the rule for `x64vfmsub213`. I didn't look much into how that works, but the first two test cases actually test this rule both in the case of passing a value in a register or a sinkable load. There, the load is definitely sunk.

Reply:
> I forget what the details of our lowering and ISLE conversions are these days, but does this need to guard on sinkable loads?

We do some checks when converting from a `Value` into `XmmMem` where we force the value into a register if it's not eligible for being sunk (i.e. has multiple uses), so this looks okay to me.

Reply:
Thanks for the clarification @afonso360 and @jameysharp!